.github/workflows/code-diff-analysis.yaml

```diff
@@ -0,0 +1,100 @@
+name: "Code diff analysis"
```

Comment:
Would name this different if possible
small nitpick since this will last a long time.
Would want to differentiate between the other Code Analysis workflow we already have
https://github.com/tenstorrent/tt-metal/actions/workflows/code-analysis.yaml

Reply:
in my mind ... code-analysis means we are running clang-tidy

code-diff-analysis means we are running clang-tidy-diff

the names are cool with me.

what would you prefer?